### DISCUSSION TITLE
Change detection in classes

### DISCUSSION THREAD

**Discussion by woubuc** [adonisjs/core#3569] [Help]
I have some JSON fields in my database and I would like to use the `prepare` and `consume` hooks to convert this JSON data into a custom class in my codebase.

```ts
@column({
  consume: json => new MyClass(json),
  prepare: c => c.toJson(),
})
myField: MyClass;
```

However, the change detection in Adonis does not pick up on any changes inside this class. The class instance doesn't change, only internal data changes, so Adonis seems to consider them equal and doesn't update the column in the database.

Can I do something (set a property on the entity, or perhaps provide a getter in `MyClass`) to tell Adonis when the value has changed so it gets saved?

**Comment by thetutlage** [comment]
Yeah, something that should be handled by Lucid. I do not have a good answer right now. However, I will turn this into an issue and work on it later
